optuna/visualization/_utils.py

```diff
@@ -156,8 +156,15 @@ def _target(t: FrozenTrial) -> float:
 
     filtered_trials: List[FrozenTrial] = []
     for trial in trials:
+        value = target(trial)
+
+        if not isinstance(value, float):
+            raise ValueError(
+                f"Trial{trial.number} has COMPLETE state, but its target value is non-numeric."
```

Comment:
`_filter_nonfinite` function does not ensure that trials always have the COMPLETE state. The current implementation filters incomplete trials when calling this function, but we'll need to synchronize the code if we hard-code the state. How about using the trial state as follows?
```suggestion
                f"Trial{trial.number} has {trial.state.name} state, but its target value is non-numeric."
```

Reply:
Thanks. I suppose the state information looks unnecessary to show the error message, so I've modified a message by considering https://github.com/optuna/optuna/pull/3768#discussion_r919724822.

Reply:
Thanks!